File: solution/1496. Path Crossing/1496.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        set<pair<int, int>> P;
        int x = 0, y = 0;
        P.insert({x, y});
        for (char& dir : path) {
            if (dir == 'N')
                ++y;
            else if (dir == 'S')
                --y;
            else if (dir == 'E')
                ++x;
            else if (dir == 'W')
                --x;
            if (P.count({x, y}))
                return true;
            P.insert({x, y});
        }
        return false;
    }
};

File: solution/1496. Path Crossing/1496.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        x = y = 0
        seen = {(x, y)}
        for d in path:
            if d == 'N':
                y += 1
            elif d == 'S':
                y -= 1
            elif d == 'E':
                x += 1
            elif d == 'W':
                x -= 1
            if (x, y) in seen:
                return True
            seen.add((x, y))
        return False
    
